registry/storage/blobstore.go

```diff
@@ -188,10 +349,50 @@ func (bs *blobStatter) Stat(ctx context.Context, dgst digest.Digest) (distributi
 	}, nil
 }
 
+// Clear deletes blob's data with or without its parent directory depending on
+// bs' configuration.
 func (bs *blobStatter) Clear(ctx context.Context, dgst digest.Digest) error {
-	return distribution.ErrUnsupported
+	var (
```

Comment:
Ok, moved the `blobStore.Clear()` implementation to `blobStore.Delete()`. `linkedBlobStore` kept as it is though.